frontend/views/containers/settings/UserProfile.vue

```diff
@@ -160,53 +138,59 @@ export default {
     ]),
     attributes () {
       return this.$store.getters.ourUserIdentityContract.attributes || {}
-    },
-    userPictureInitial () {
-      return this.attributes.picture
     }
   },
   methods: {
     openModal (mode) {
       sbp('okTurtles.events/emit', OPEN_MODAL, mode)
       return false
     },
-    async save () {
-      if (this.ephemeral.newPicture) {
-        try {
-          this.form.picture = await imageUpload(this.form.picture)
-          this.ephemeral.newPicture = false
-        } catch (error) {
-          console.error(error)
-          this.ephemeral.errorMsg = L('Failed to upload user picture')
-          return false
-        }
+    async fileChange (fileList) {
+      if (!fileList.length) return
+      const fileReceived = fileList[0]
+      let picture
+
+      try {
+        picture = await imageUpload(fileReceived)
+      } catch (error) {
+        console.error(error)
+        this.$refs.pictureMsg.danger(`${L('Failed to upload avatar. Please, try again.')} ${error.message}`)
+        return false
       }
 
       try {
-        this.ephemeral.profileSaved = false
-        const attrs = {}
-        for (const key in this.form) {
-          if (this.form[key] !== this.attributes[key]) {
-            attrs[key] = this.form[key]
-          }
+        await this.setAttributes({ picture })
+        this.$refs.picture.setFromBlob(fileReceived)
+        this.$refs.pictureMsg.success(L('Picture updated!'))
+      } catch (error) {
+        console.error(error)
+        this.$refs.pictureMsg.danger(`${L('Failed to upload picture.')} ${error.message}`)
```

Comment:
`${L('Failed to upload picture.')} ${error.message}` => `L('Failed to upload picture: {error}', { error: error.message })`